chain/types/actor_event.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// The value codec to match when filtering event values.
+	Codec uint64 `json:"codec"`
+
+	// The value to want to match on associated with the corresponding "event key"
+	// when filtering events.
+	// Should be a byte array encoded with the specified codec.
+	// Assumes base64 encoding when converting to/from JSON strings.
+	Value []byte `json:"value"`
+}
+
+type SubActorEventFilter struct {
+	Filter ActorEventFilter `json:"filter"`
+
+	// If true, all available matching historical events will be written to the response stream
+	// before any new real-time events that match the given filter are written.
+	Prefill bool `json:"prefill"`
```

Comment:
We already have `FromBlock` in the filter, so we have two ways to signal a similar thing. So we can have a situation like `Prefill=true` + `FromBlock="latest"`, or `Prefill=false` + `FromBlock="earliest"`. In fact, the only really meaningful value of `FromBlock` with a `Prefill=false` would be a future epoch, all other possible values yield the same result.

So I wonder if we should just remove `SubActorEventFilter` entirely and rely on `FromBlock` to do the prefilling?

Reply:
I agree, I've removed the `SubActorEventFilter `. We will now Prefill if FromEpoch < Current Epoch.

Reply:
@rvagg After making this change and writing a test to see how it feels, I think it's best if we keep this explicit for the user so they know what they're asking for.

I am adding some docs to point out the edge case you mention here.